presto-main/src/main/java/com/facebook/presto/operator/scalar/UnitFunctions.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/**
+ * UDFs for reporting values in succinct representations
+ */
+public final class UnitFunctions
+{
+    private UnitFunctions()
+    {
+    }
+
+    @Description("succinct duration string")
+    @ScalarFunction("succinct_duration")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctDuration(@SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.VARCHAR) Slice unit)
```

Comment:
It's good practice to only wrap the relevant code that can throw with try/catch:

```
Duration duration;
try {
    duration = Duration.succinctDuration(value, valueOfTimeUnit(unit.toStringUtf8()));
}
catch (IllegalArgumentException e) {
    // When value is negative, NaN, etc.
    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
}
return utf8Slice(duration.toString());
```